core/src/test/scala/integration/kafka/api/SaslClientsWithInvalidCredentialsTest.scala

```diff
@@ -248,4 +250,25 @@ class SaslClientsWithInvalidCredentialsTest extends IntegrationTestHarness with
     producerConfig.put(ProducerConfig.ENABLE_IDEMPOTENCE_CONFIG, "true")
     createProducer()
   }
+
+  private def createScramAdminClient(user: String, password: String): Admin = {
```

Comment:
It feels like we are accumulating a lot of these "create an admin client, but with SCRAM" functions.  Since all these tests ultimately subclass SaslSetup, can't we have a common function there?

Reply:
Ok, I added the following code to SaslSetup, and we implement that first method in the 3 test classes that use this functionality.

```
  def createPrivilegedAdminClient(): Admin = {
    // create an admin client instance that is authorized to create credentials
    throw new UnsupportedOperationException("Must implement this if a test needs to use it")
  }

  def createScramCredentialsViaPrivilegedAdminClient(userName: String, password: String): Unit = {
    val privilegedAdminClient = createPrivilegedAdminClient() // must explicitly implement this method
    try {
      // create the SCRAM credential for the given user
      createScramCredentials(privilegedAdminClient, userName, password)
    } finally {
      privilegedAdminClient.close()
    }
  }
```